protocol/messenger_handler.go

```diff
@@ -1146,8 +1150,22 @@ func (m *Messenger) HandleCommunityRequestToJoinResponse(state *ReceivedMessageS
 			return err
 		}
 		if len(response.Communities()) > 0 {
-			state.Response.AddCommunity(response.Communities()[0])
-			state.Response.AddCommunitySettings(response.CommunitiesSettings()[0])
+			communitySettings := response.CommunitiesSettings()[0]
+			community := response.Communities()[0]
+			state.Response.AddCommunity(community)
+			state.Response.AddCommunitySettings(communitySettings)
+
+			magnetlink := requestToJoinResponseProto.MagnetUri
+			if m.torrentClientReady() && communitySettings != nil && communitySettings.HistoryArchiveSupportEnabled && magnetlink != "" {
+				m.downloadHistoryArchiveTasksWaitGroup.Add(1)
+				go func() {
+					defer m.downloadHistoryArchiveTasksWaitGroup.Done()
+					m.downloadAndImportHistoryArchives(community.ID(), magnetlink)
```

Comment:
What happens if we receive the message twice? does it start two separate tasks?

Reply:
So here the same mechanism kicks in as mentioned in: https://github.com/status-im/status-go/pull/3013/files#r1045551410

essentially `downloadAndImportHistoryArchives()` will first cancel any ongoing download/importing tasks for this community, then start its own.